src/app/components/application/application-form/application-form.component.ts

```diff
@@ -900,71 +904,95 @@ export class ApplicationFormComponent implements OnInit {
+    appObj.treeAutoRefresh = this.applicationForm.value.treeAutoRefresh;
+    appObj._links = this.applicationForm.value._links;
+    appObj.situationMap = situationMap;
+    if (this.applicationID == -1) {
+      appObj.id = null;
+    } else {
+      appObj.id = this.applicationForm.value.id;
+      appObj.createdDate = this.applicationToEdit.createdDate
+    }
 
-      this.applicationService.save(appObj)
-        .subscribe(async resp => {
 
-          this.applicationToEdit = resp;
-          this.applicationSaved = resp;
-          this.applicationID = this.applicationToEdit.id;
-          this.applicationForm.patchValue({
-            id: resp.id,
-            _links: resp._links
-          })
+    this.applicationService.save(appObj)
+      .subscribe(async resp => {
 
-          this.utils.saveTranslation(resp.id, this.nameTranslationMap, this.applicationToEdit.name, this.nameTranslationsModified);
-          this.nameTranslationsModified = false;
-          this.utils.saveTranslation(resp.id, this.titleTranslationMap, this.applicationToEdit.title, this.titleTranslationsModified);
-          this.titleTranslationsModified = false;
-
-          this.getAllElementsEventParameters.next("save");
-          this.getAllElementsEventTemplateConfiguration.next("save");
-          this.getAllElementsEventRoles.next("save");
-          this.getAllElementsEventBackground.next("save");
-          this.getAllElementsEventTree.next("save");
-        },
-          error => {
-            console.log(error)
-          });
-    }
-    else {
-      this.utils.showRequiredFieldsError();
-    }
+        this.applicationToEdit = resp;
+        this.applicationSaved = resp;
+        this.applicationID = this.applicationToEdit.id;
+        this.applicationForm.patchValue({
+          id: resp.id,
+          _links: resp._links
+        })
 
+        this.utils.saveTranslation(resp.id, this.nameTranslationMap, this.applicationToEdit.name, this.nameTranslationsModified);
+        this.nameTranslationsModified = false;
+        this.utils.saveTranslation(resp.id, this.titleTranslationMap, this.applicationToEdit.title, this.titleTranslationsModified);
+        this.titleTranslationsModified = false;
 
+        this.getAllElementsEventParameters.next("save");
+        this.getAllElementsEventTemplateConfiguration.next("save");
+        this.getAllElementsEventRoles.next("save");
+        this.getAllElementsEventBackground.next("save");
+        this.getAllElementsEventTree.next("save");
+      },
+        error => {
+          console.log(error)
+        });
+  }
+
+  validTrees() {
+    let valid = true;
+    const trees = this.treesDataGrid.rowData;
+    valid = this.validTreesType(trees);
+    return valid;
   }
 
+  validTreesType(trees) {
+    let valid = true;
+    const filterTrees = trees.filter(a => a.status !== 'pendingDelete')
+    if (this.currentAppType === 'app-turistica') {
```

Comment:
These values that are linked to code lists should be defined as constants in a separate file and documented.